server/src/main/java/io/crate/planner/optimizer/rule/MergeFilterAndCollect.java

```diff
@@ -49,6 +49,11 @@ public MergeFilterAndCollect() {
             .with(source(), typeOf(Collect.class).capturedAs(collectCapture));
     }
 
+    @Override
+    public boolean canBeConfiguredBySessionSetting() {
+        return false;
```

Comment:
I think it would be good to have comments that explain _why_ the rules are mandatory.

Reply:
👍